src/main/java/com/exadel/discount/service/impl/CityServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package com.exadel.discount.service.impl;
+
+import com.exadel.discount.dto.CityDTO;
+import com.exadel.discount.entity.City;
+import com.exadel.discount.exception.NotFoundException;
+import com.exadel.discount.mapper.CityMapper;
+import com.exadel.discount.repository.CityRepository;
+import com.exadel.discount.repository.CountryRepository;
+import com.exadel.discount.service.CityService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CityServiceImpl implements CityService {
+
+    private final CityRepository cityRepository;
+    private final CityMapper cityMapper;
+    private final CountryRepository countryRepository;
+
+    @Override
+    public List<CityDTO> findAll() {
+        log.debug("Getting list of all Cities");
+
+        List<CityDTO> cityDTOList = cityMapper.getListDTO(cityRepository.findAll());
+
+        log.debug("Successfully got list of all Cities");
+        return cityDTOList;
+    }
+
+    @Override
+    public List<CityDTO> findAllByCountryName(String countryName) {
+        log.debug("Getting list of all Cities by CountryName");
+
+        List<CityDTO> cityDTOList = cityMapper.getListDTO(cityRepository.findByCountryName(countryName));
+
+        log.debug("Successfully got list of all Cities by CountryName");
+        return cityDTOList;
+    }
+
+    @Override
+    public List<CityDTO> findAllByCountryId(UUID countryId) {
+        log.debug("Getting list of all Cities by CountryId");
+
```

Comment:
please remove the empty line here or keep the same style for logging in other methods

Reply:
Done